### PR TITLE
Adding rsa signing functions

### PR DESCRIPTION
**Pull request by themetaschemer**
Extending this package to include 3 new signing functions: RS256, RS384 and RS512. 

I am using the pk-sign procedure provided by the crypto module. Had to tweak the signatures of the high-level interface to allow bytes for secrets.

### CODE REVIEW

**Comment by RenaissanceBug** [comment]
Thanks! I'll review & respond within the next few days.

**Comment by themetaschemer** [comment]
Hi, just checking in to see if you had a chance to review. Thanks.

**Comment by RenaissanceBug** [comment]
Thanks for the nudge. Will look right now.

File: net/jwt/algorithms.rkt [hunk 1]
```diff
@@ -22,7 +33,8 @@
          current-string-converter
          SigningFunction
          ok-signature?
-         none hs256 hs384 hs512
+         none hs256 hs384 hs512 rs256
```
Comment (RenaissanceBug) [inline net/jwt/algorithms.rkt]:
```suggestion
         none hs256 hs384 hs512
```

**Review by RenaissanceBug — CHANGES_REQUESTED** [review]
Looks good! thanks!

**Comment by themetaschemer** [comment]
Cool. Accepted the suggested change.

**Comment by RenaissanceBug** [comment]
All done. Thanks again (for the changes, and your patience!).

**Comment by themetaschemer** [comment]
Thanks! We've made using Racket a little bit better :)

**Comment by RenaissanceBug** [comment]
Yay!

BTW, this library started as an offshoot of a web project that I’ve long since abandoned, and (now coming back from a 2-year sabbatical from coding) I don’t have a strong sense of what functionality is important to implement next. If you have encountered other wants as a user of it, please let me know.

Thanks again.

**Comment by themetaschemer** [comment]
Yes, will definitely do that. I actually started using this library to implement authentication for Google's cloud services. I picked up another abandoned google cloud api project and updated it to use their modern jwt based auth for service accounts. I ultimate goal would be to evolve this library into a full Google cloud services api library using the machine-readable api spec that Google has. But your jwt library is the foundation for the auth in it!
A. 
    On Friday, December 4, 2020, 12:18:30 PM PST, Jordan Johnson <notifications@github.com> wrote:  
 
 

Yay!

BTW, this library started as an offshoot of a web project that I’ve long since abandoned, and (now coming back from a 2-year sabbatical from coding) I don’t have a strong sense of what functionality is important to implement next. If you have encountered other wants as a user of it, please let me know.

Thanks again.

**Comment by RenaissanceBug** [comment]
Oh, that’s fantastic! It was the need for authentication that prompted me to start working on the lib, so I’m delighted that the actual auth machinery is now getting built on top of it. 😁

**Pull request merged**